src/main/scala/epic/parser/models/NeuralParserTrainer.scala

```diff
@@ -0,0 +1,241 @@
+package epic.parser.models
+
+import java.io.File
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+
+import breeze.config.Help
+import breeze.linalg._
+import breeze.optimize._
+import breeze.optimize.FirstOrderMinimizer.OptParams
+import breeze.util._
+import breeze.util.Implicits._
+import epic.constraints.CachedChartConstraintsFactory
+import epic.constraints.ChartConstraints
+import epic.dense.AdadeltaGradientDescentDVD
+import epic.framework._
+import epic.parser._
+import epic.parser.ParseEval.Statistics
+import epic.parser.ParserParams.XbarGrammar
+import epic.parser.projections.OracleParser
+import epic.parser.projections.ParserChartConstraintsFactory
+import epic.trees.AnnotatedLabel
+import epic.trees.TreeInstance
+import epic.trees.annotations._
+import epic.util.CacheBroker
+
+
+/**
+ * The main entry point for training discriminative parsers.
+ * Has a main method inherited from ParserPipeline.
+ * Use --help to see options, or just look at the Params class.
+ *
+ *
+ */
+object NeuralParserTrainer extends epic.parser.ParserPipeline with LazyLogging {
```

Comment:
Thanks :) I wanted to keep ParserTrainer unchanged from what it was before. It was also full of casts that wouldn't be that easy to get rid of.